docs/proposals/konnectivity-openvpn.md

```diff
@@ -0,0 +1,120 @@
+
+* In case that the service account token volume projection is not yet enabled (which may be already done as part of [this issue][issue_6191]), we need to add some more kube-apiserver flags if Konnectivity is enabled.
+* In case that service account token volume projection is already enabled, we need to add `system:konnectivity-server` into `api-audiences` list.
+
+Apart from that, the API Server needs the `--egress-selector-config-file` flag pointed to the file with `EgressSelectorConfiguration`, which defines the type of connection between the API Server and the Konnectivity Server. 
+
+### Konnectivity Server in Control Plane Cluster
+
+The Konnectivity server runs in the control plane cluster and acts as a proxy between the Kube API server running in the same cluster and Konnectivity Agents that are running in the user cluster and are connecting to the Konnectivity Server’s IP address, which needs to be reachable from the user cluster (for this connection, Kubermatic uses various [Expose Strategies][k8c_expose_strategies]).
+
+Konnectivity server will be deployed as a sidecar container for the Kubernetes API Server. Connection between the two will be done by using UDS.
+
+
+### Konnectivity Agents in User Cluster
+
+The Konnectivity agents run in the user cluster and are connecting to the specified Konnectivity server IP address. Since it is not necessary to have Konnectivity agents running on each node in the user cluster, they can be deployed as a Deployment with fewer replicas than the number of nodes. The exact number of replicas can be configurable on a per-cluster basis, with e.g. 2 as a default.
+
+If there was a need for that going forward, we can potentially add a controller which would dynamically scale the number of replicas based on the number of nodes currently running in the cluster (e.g. if node auto-scaling is enabled).
+
+
+### Impact on cluster network reachability for other control plane components
+
+Currently, multiple control plane components (not only k8s API server, but also e.g. scheduler, controller-manager, openstack-cloud-controller-manager, etc.) are configured to be able to reach the user cluster network. The current solution is based on `openvpn-client` sidecar containers in the control plane pods, and custom `dnsConfig` pointing to a control-plane cluster-local `dns-resolver` deployment. With Konnectivity, this won’t be possible, since Konnectivity provides the cluster network connectivity only for the API server. However, we believe that given the k8s hub-and-spoke design, these components do not actually need the connectivity to anything else other than the k8s API server. If there is a component which currently runs in the control plane cluster and needs the connectivity to the cluster network, we will have to run it in the user cluster.
```

Comment:
A notable example of components that should be moved on the workers side is metrics-server, because Konnectivity does not support aggregation layers:

https://github.com/kubernetes/kubernetes/issues/92374

This is not the only limitation, metrics-server should also be extended to "pass" from Konnectivity in order to reach kubelets.

Reply:
So what you are saying right now is that the metrics-server can't be deployed on the usercluster side which leads to deploy it and make it use the Konnectivity right?

Reply:
Nope, I meant that at the moment metrics-server cannot be deployed on `seed` cluster side but only on `user` side, so we have no options but moving it.
Otherwise, we would have to reopen and try to push for kubernetes/kubernetes#92374, and contributing to metrics-server too in order to make it "Konnectivity" ready.

Reply:
Gotcha 👍🏻